### PR TITLE
Add SimpleCASTNode

### PR DESCRIPTION
**Pull request by kriw**
This issue is related to #136 

TODO:
- [x] Implement a converter to CAST from SimpleCAST
    - [x] Revise (or add methods) struct `c_simple::CAST`
- [x] Write unit tests

### CODE REVIEW

File: src/backend/lang_c/c_simple_ast.rs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    fn is_exit(&self) -> bool {
+        match self {
+            &SimpleCASTNode::Action(ActionNode::Return) => true,
+            _ => false,
+        }
+    }
+}
+
+/// Returns nodes which is connected with given type of edge
+fn neighbors_by_edge(edges: &Vec<EdgeReference<SimpleCASTEdge>>, ty: &SimpleCASTEdge) -> Vec<NodeIndex> {
```
Comment (chinmaydd) [inline src/backend/lang_c/c_simple_ast.rs]: Hey @kriw , any specific reason why this method is not part of the `impl` for SimpleCAST?

**Comment by chinmaydd** [comment]
Also, are you yet to account for type information or am I missing something?

**Comment by kriw** [comment]
Yes, but I'm wondering how to implement it.
Currently, I'm planning to add type field to `ValueNode::Variable` and `ValueNode::Constant`.

**Comment by chinmaydd** [comment]
Apart from a few naming conventions here and there(Eg. `add_goto() and insert_goto()` dont really convey the complete meaning to the API user. But since we are not in v1 we can always make changes later) this LGTM @kriw . Good work :+1:  The `SimpleCAST` is compact and incorporates the logic we need.

P.S: I think we should implement `fmt::Debug` for CAST rather than having the `print()` method for it. Might be more helpful.

**Comment by sushant94** [comment]
This seems good to me! Awesome work @kriw! :)

**Pull request merged**